src/combineValidators.ts

```diff
@@ -0,0 +1,21 @@
+import { Validator } from "./types";
+
+export const combineValidators =
+  <Value, ErrorMessage = string>(
+    ...validators: (Validator<Value, ErrorMessage> | false)[]
+  ): Validator<Value, ErrorMessage> =>
+  (value) => {
+    const [validator, ...nextValidators] = validators;
+
+    if (validator != null && validator !== false) {
+      const result = validator(value);
+
+      if (typeof result !== "undefined") {
+        return result;
+      }
+    }
+
+    if (nextValidators.length > 0) {
+      return combineValidators(...nextValidators)(value);
+    }
```

Comment:
shouldn't we use a `while` loop rather than recursion here? most browsers don't have TCO

Reply:
Support is actually quite decent: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#browser_compatibility

Reply:
And it's transpiled to:

```
const r = function () {
	return (e) => {
		const [t, ...n] = [].slice.call(arguments);
		if (null != t && !1 !== t) {
			const r = t(e);
			if (void 0 !== r) return r;
		}
		if (n.length > 0) return r(...n)(e);
	};
}
```

Reply:
Not sure this is an issue as you will probably never chain a lot of validators. If TCO is a must this would mean we have thousand of them chained (it was already like this in current version)